src/validation.ts

```diff
@@ -96,7 +96,10 @@ export class Validator {
             joi.validate(target, classSchema, options) :
             joi.validate(target, classSchema);
 
-        return result as ValidationResult<TInstance>;
+        return {
+            ...result,
+            value: result.value as TInstance,
```

Comment:
P.S. Here we're pretending that `Partial<TInstance>` is actually a `TInstance`. Same below.